src/Filters/remez_fir.jl

```diff
@@ -0,0 +1,845 @@
+OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+THE POSSIBILITY OF SUCH DAMAGE.
+
+--- end of scipy license
+
+The remez and associated functions were extracted from sigtoolsmodule.c,
+with the following header comment:
+
+SIGTOOLS module by Travis Oliphant
+
+Copyright 2005 Travis Oliphant
+Permission to use, copy, modify, and distribute this software without fee
+is granted under the SciPy License.
+
+C CODE BANNER
+
+/********************************************************
+ *
+ *  Code taken from remez.c by Erik Kvaleberg which was 
+ *    converted from an original FORTRAN by:
+ *
+ * AUTHORS: JAMES H. MCCLELLAN
+ *
+ *         DEPARTMENT OF ELECTRICAL ENGINEERING AND COMPUTER SCIENCE
+ *         MASSACHUSETTS INSTITUTE OF TECHNOLOGY
+ *         CAMBRIDGE, MASS. 02139
+ *
+ *         THOMAS W. PARKS
+ *         DEPARTMENT OF ELECTRICAL ENGINEERING
+ *         RICE UNIVERSITY
+ *         HOUSTON, TEXAS 77001
+ *
+ *         LAWRENCE R. RABINER
+ *         BELL LABORATORIES
+ *         MURRAY HILL, NEW JERSEY 07974
+ *
+ *  
+ *  Adaptation to C by 
+ *      egil kvaleberg
+ *      husebybakken 14a
+ *      0379 oslo, norway
+ *  Email:
+ *      egil@kvaleberg.no
+ *  Web:
+ *      http://www.kvaleberg.com/
+ *
+ *********************************************************/
+
+=============================================#
+
+"""/*
+ *-----------------------------------------------------------------------
+ * FUNCTION: lagrange_interp (d)
+ *  FUNCTION TO CALCULATE THE LAGRANGE INTERPOLATION
+ *  COEFFICIENTS FOR USE IN THE FUNCTION gee.
+ *-----------------------------------------------------------------------
+ */"""
+function lagrange_interp(k::Integer, n::Integer, m::Integer, x::AbstractVector)
+    retval = 1.0;
+    q = x[k];
+    for l = 1 : m
+        for j = l : m : n
+            if (j != k)
+                retval *= 2.0 * (q - x[j])
+            end
+        end
+    end
+    1.0 / retval
+end
+
+
+"""
+eff(freq::Float64, fx::AbstractVector, lband::Integer, jtype::Integer)
+
+/*
+ *-----------------------------------------------------------------------
+ * FUNCTION: eff
+ *  FUNCTION TO CALCULATE THE DESIRED MAGNITUDE RESPONSE
+ *  AS A FUNCTION OF FREQUENCY.
+ *  AN ARBITRARY FUNCTION OF FREQUENCY CAN BE
+ *  APPROXIMATED IF THE USER REPLACES THIS FUNCTION
+ *  WITH THE APPROPRIATE CODE TO EVALUATE THE IDEAL
+ *  MAGNITUDE.  NOTE THAT THE PARAMETER FREQ IS THE
+ *  VALUE OF NORMALIZED FREQUENCY NEEDED FOR EVALUATION.
+ *-----------------------------------------------------------------------
+ */
+"""
+function eff(freq::Float64, fx::AbstractVector, lband::Integer, jtype::Integer)
+    if (jtype != 2)
```

Comment:
Done, with exceptions.
There are many single-line "if" statements, e.g.
```        if (l > 1) l = l-1; end```
I've left these in place - we can expand these out but it will expand the code quite a bit.

Reply:
those could be replaced with short-circuiting, e.g. in this case `l > 1 && (l = l-1)` IMO `cond && expr` is more idiomatic julia for one-line conditionals than `if cond expr; end`